pkg/controller/apiservicecabundle/starter.go

```diff
@@ -0,0 +1,48 @@
+package apiservicecabundle
+
+import (
+	"fmt"
+	"io/ioutil"
+	"time"
+
+	"k8s.io/client-go/rest"
+	apiserviceclient "k8s.io/kube-aggregator/pkg/client/clientset_generated/clientset"
+	apiserviceinformer "k8s.io/kube-aggregator/pkg/client/informers/externalversions"
+
+	servicecertsignerv1alpha1 "github.com/openshift/api/servicecertsigner/v1alpha1"
+)
+
+type APIServiceCABundleInjectorOptions struct {
+	Config *servicecertsignerv1alpha1.APIServiceCABundleInjectorConfig
+}
+
+func (o *APIServiceCABundleInjectorOptions) RunAPIServiceCABundleInjector(clientConfig *rest.Config, stopCh <-chan struct{}) error {
+	apiServiceClient, err := apiserviceclient.NewForConfig(clientConfig)
+	if err != nil {
+		return err
+	}
+	apiServiceInformers := apiserviceinformer.NewSharedInformerFactory(apiServiceClient, 2*time.Minute)
+
+	caBundleFile := o.Config.CABundleFile
+	if len(caBundleFile) == 0 {
+		return fmt.Errorf("no signing cert/key pair provided")
+	}
+	caBundleContent, err := ioutil.ReadFile(caBundleFile)
```

Comment:
i assume the caBundleFile will be provided by the secret, right? can't we have go routine that is just periodically checking the content of that file and trigger the controller if the file change on disk? (we don't need to have configMap triggers then I guess?)

Reply:
> i assume the caBundleFile will be provided by the secret, right? can't we have go routine that is just periodically checking the content of that file and trigger the controller if the file change on disk? (we don't need to have configMap triggers then I guess?)

I plan to write a health check that will detect file changes and fail.